test/ruby_lsp_rails/rails_test_style_test.rb

```diff
@@ -0,0 +1,176 @@
+# typed: true
+# frozen_string_literal: true
+
+require "test_helper"
+
+module RubyLsp
+  module Rails
+    class RailsTestStyleTest < ActiveSupport::TestCase
+      test "discovers rails declarative tests" do
+        source = <<~RUBY
+          class SampleTest < ActiveSupport::TestCase
+            test "first test" do
+              assert true
+            end
+
+            test "second test" do
+              assert true
+            end
+          end
+        RUBY
+
+        with_active_support_declarative_tests(source) do |items|
+          assert_equal(1, items.length)
+          test_class = items.first
+          assert_equal("SampleTest", test_class[:label])
+          assert_equal(2, test_class[:children].length)
+
+          test_labels = test_class[:children].map { |i| i[:label] }
+          assert_includes(test_labels, "first test")
+          assert_includes(test_labels, "second test")
+          assert_all_items_tagged_with(items, :rails)
+        end
+      end
+
+      test "discovers rails test with empty test name" do
```

Comment:
Does Rails prevent you from running a test with an empty name? I never tried it, so just confirming if we're choosing the right behaviour.